app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -35,14 +36,19 @@ def context_urls
 
       def smart_proxy
         @smart_proxy ||= begin
-          proxy = params[:url] ? SmartProxy.unscoped.find_by(url: params[:url]) : SmartProxy.pulp_primary
+          proxy = params[:url] ? find_smart_proxy_by_url : SmartProxy.pulp_primary
 
           fail Foreman::Exception, _('Smart proxy content source not found!') unless proxy
           fail Foreman::Exception, _('Pulp 3 is not enabled on Smart proxy!') unless proxy.pulp3_enabled?
 
           proxy
         end
       end
+
+      def find_smart_proxy_by_url
+        auth_smart_proxy
+        return @detected_proxy if @detected_proxy
```

Comment:
I think it's sufficient to use
```suggestion
        @detected_proxy
```